src/api/app/components/workflow_run_detail_component.html.haml

```diff
@@ -35,7 +35,20 @@
         = pretty_request_payload
   .tab-pane.fade{ id: "workflow-configuration-tab-content#{id}", role: 'tabpanel',
                   aria: { labelledby: "workflow-configuration-tab#{id}" } }
-    = workflow_configuration_data
+    - if configuration_source_not_available
```

Comment:
How can you have a `WorkflowRun` from running a `Workflow` without a configuration? :)

Reply:
This check is necessary for those old workflow runs that didn't store any of those values: `workflow_configuration`, `workflow_configuration_path` or `workflow_configuration_url`. When we introduced them it wasn't possible to backfill those values.

From that moment on, those values are no longer empty unless they aren't downloadable (we create the workflow run before we check). See this line and other similar ones in the same file:
https://github.com/openSUSE/open-build-service/blob/cfe9121d0dbee5e2643f88278e6ad0f226a1898f/src/api/app/services/workflows/yaml_downloader.rb#L74

Reply:
The code has been refactored, but it still checks the existence of those values upfront.